ci: add golangci-lint to the GitHub action

fcuny-rbx opened the pull request:
golangci-lint is a linter for the Go programming language. It runs many
tools and linter, for example staticcheck, which, using static analysis,
finds bugs and performance issues, offers simplifications, and enforces
style rules.

Making it part of the CI path with help with enforcing good standards
and follow the community style rules.

staticcheck is also added to the Vagrant configuration so it can be run
locally before creating a pull request.

fcuny-rbx commented:
This depends on https://github.com/Roblox/nomad-node-problem-detector/pull/53

fcuny-rbx commented:
Hm, something will be needed here: https://github.com/Roblox/nomad-node-problem-detector/actions/runs/2693554878

shishir-a412ed commented:
@fcuny-rbx Are there instructions on how to run `staticcheck` locally in vagrant VM? 
Shall we add something in the `README.md` under this [`section`](https://github.com/Roblox/nomad-node-problem-detector#tests)

shishir-a412ed commented:
@fcuny-rbx Strangely, the CI (NNPD tests) didn't run on the PR for some reason. Can we rebase (and repush) and see if it triggers the CI?

fcuny-rbx commented:
@shishir-a412ed I've rebased, and switched to our own github action for golang-ci-check, and now the actions are passing. Let me know if I'm missing something else.

shishir-a412ed commented:
<img width="1188" alt="Screen Shot 2022-08-31 at 8 24 46 AM" src="https://user-images.githubusercontent.com/4207652/187717230-c2338e64-de50-44e8-bed2-b75a790e0b9c.png">

@fcuny-rbx Shall we rename the `NNPD-CI / Run CI` to `NNPD-CI / Golang static check`?

fcuny-rbx commented:
Ah yes, good idea, it's more explicit - fixing this.

shishir-a412ed reviewed (approved):
Small nitpick, otherwise LGTM!
